pkg/slices/slices.go

```diff
@@ -149,3 +149,17 @@ func XorNil[T any](s1, s2 []T) bool {
 	return s1 == nil && s2 != nil ||
 		s1 != nil && s2 == nil
 }
+
+// True returns true if s is not empty and pred is true for each element in s,
```

Comment:
This makes the function quite specific by checking the length of the slice, what do you think about returning True even though the slice is empty? It would allow broader re-usability. 

Let's rename it to something like `AllMatch`

Maybe let's also include something like
```
May not evaluate on all elements if not necessary for determining the result. If slice is empty then true is returned and predicate is not evaluated.
```

Reply:
Yep, I can easily move the check from where it is called (it is needed in this use case).
Regarding the name I though about `All` because of the similarity with Python `all()` built-in, but ended up using `True` because of https://pkg.go.dev/slices#All.
But I agree that `AllMatch` might be better, let's change it! :+1:

Reply:
Oh, thanks, I missed the check on len(links). Let's remove it, then!